membership-attribute-service/app/controllers/AccountController.scala

```diff
@@ -271,6 +273,40 @@ class AccountController(commonActions: CommonActions, override val controllerCom
     }
   }
 
+
+  def allSubscriptions(contactRepo: SimpleContactRepository, subService: SubscriptionService[Future])(maybeUserId: Option[String]): OptionT[OptionEither.FutureEither, List[Subscription[SubscriptionPlan.AnyPlan]]] = for {
+    user <- OptionEither.liftFutureEither(maybeUserId)
+    contact <- OptionEither(contactRepo.get(user))
+    subscriptions <- OptionEither.liftEitherOption(subService.current[SubscriptionPlan.AnyPlan](contact)) // TODO are we happy with an empty list in case of error ?!?!
+  } yield subscriptions
+
+  private def allPaymentDetails = BackendFromCookieAction.async { implicit request =>
+    implicit val tp = request.touchpoint
+    def getPaymentMethod(id: PaymentMethodId) = tp.zuoraRestService.getPaymentMethod(id.get)
+    val maybeUserId = authenticationService.userId
+
+    logger.info(s"Attempting to retrieve payment details for identity user: ${maybeUserId.mkString}")
+    (for {
```

Comment:
for other reviewers - I think this line to 294 is the only part of `allPaymentDetails` that isn't just copy and paste from the soon to be deprecated version of the function